src/test/java/org/springframework/data/couchbase/repository/query/support/N1qlUtilsTest.java

```diff
@@ -148,4 +151,18 @@ public void testCreateWhereFilterForEntityWithBaseCriteria() throws Exception {
 
     assertEquals(expected, real);
   }
+
+  @Test
+  public void testCreateWhereFilterForEntityTakesTypeAliasAnnotationIntoAccount() throws Exception {
+    String expected = "`_class` = \"CustomDiscriminator\"";
+    CouchbaseConverter converter = mock(CouchbaseConverter.class);
+    when(converter.getTypeKey()).thenReturn("_class");
+    EntityMetadata metadata = mock(EntityMetadata.class);
+    @TypeAlias("CustomDiscriminator") class EntityMock {};
+    when(metadata.getJavaType()).thenReturn(EntityMock.class);
+
+    String real = N1qlUtils.createWhereFilterForEntity(null, converter, metadata).toString();
+
+    assertEquals(expected, real);
+  }
```

Comment:
I added this test for the **if** branch of my private method `N1qlUtils.getTypeValue(EntityMetadata<?> entityInformation)`. The **else** branch (which) has been already tests by the preexisting unit tests